Library/Homebrew/requirements/gpg2_requirement.rb

```diff
@@ -1,12 +1,31 @@
 require "requirement"
-require "gpg"
 
 class GPG2Requirement < Requirement
   fatal true
   default_formula "gnupg"
 
-  # GPGTools installs GnuPG 2.0.x as a `gpg` symlink pointing
-  # to `gpg2`. Our `gnupg` installs only a non-symlink `gpg`.
   # The aim is to retain support for any version above 2.0.
-  satisfy(build_env: false) { Gpg.available? }
+  satisfy(build_env: false) do
+    which_all("gpg").detect do |gpg|
+      gpg_short_version = Utils.popen_read(gpg, "--version")[/\d\.\d/, 0]
+      next unless gpg_short_version
+      gpg_version = Version.create(gpg_short_version.to_s)
+      @version = gpg_version
+      gpg_version >= Version.create("2.0")
+    end
+  end
+
+  env do
+    # If Homebrew's GPG is installed, prioritise it.
+    ENV.prepend_path "PATH", Formula["gnupg"].opt_bin
```

Comment:
Why is that? Will this not contradict the PATH you're using for `satisfy`?

Reply:
This is something I might drop. My primary concern is things hardcoding the Cellar path to `gpg`, which would become a pain around updates/revision bumps/etc.

ILZ said `/usr/local/bin` is rejected even if it's where `satisfy` points to? I think? Unless I'm misunderstanding his words there, which is why I inserted the `append` line below.

Reply:
> My primary concern is things hardcoding the Cellar path to gpg

How would they get that?



> ILZ said /usr/local/bin is rejected even if it's where satisfy points to? I think? Unless I'm misunderstanding his words there, which is why I inserted the append line below.

Let's pass on it unless you've got a reproducible case in which it's needed.

Reply:
I was referring to 5de7ec394bf9f24423e0b380b39c6c2801eca81e.

Reply:
At that point: why don't we just `depends_on "gpg"`?

Reply:
The point was to not trample all over MacGPG2, but that's _somewhat_ less of an issue now you can permanently `brew unlink` things. It would still be an issue the first time it happened, but. Ultimately it comes down to the question of how broad a use case are you willing to accommodate from a Homebrew point of view.

Reply:
Er, GPGTools even 😅.

Reply:
> How would they get that?

This was something I stumbled upon in testing but can't reproduce now, so 🤷‍♂️.